tests/crypto/test_cases/test_aes_cbc_mac.c

```diff
@@ -0,0 +1,228 @@
+{
+	return mbedtls_cipher_set_padding_mode(ctx, padding);
+}
+
+static int cipher_crypt(mbedtls_cipher_context_t *p_ctx, size_t iv_len,
+			size_t input_len)
+{
+	size_t crypt_len = 0; /* Unused */
+
+	return mbedtls_cipher_crypt(p_ctx, m_aes_iv_buf, iv_len,
+				    m_aes_input_buf, input_len,
+				    m_aes_output_buf, &crypt_len);
+}
+
+static int _cipher_update_iterative_full_blocks(mbedtls_cipher_context_t *p_ctx,
+						size_t input_len)
+{
+	int err_code;
+	size_t round = 0;
+	size_t output_len;
+
+	do {
+		err_code = mbedtls_cipher_update(
+			p_ctx, &m_aes_input_buf[16 * round], 16,
+			&m_aes_output_buf[16 * round], &output_len);
+		round++;
+		if (dbg_hexdump_on) {
+			LOG_DBG("Output len %d", output_len);
+		}
+	} while ((err_code == 0) && ((round * 16) < input_len));
+
+	return err_code;
+}
+
+static int cipher_update_iterative(mbedtls_cipher_context_t *p_ctx,
+				   size_t input_len)
+{
+	return _cipher_update_iterative_full_blocks(p_ctx, input_len);
+}
+
+__attribute__((noinline)) static void unhexify_aes(void)
+{
+	bool encrypt = (p_test_vector->direction == MBEDTLS_ENCRYPT);
+
+	key_len = hex2bin(p_test_vector->p_key, strlen(p_test_vector->p_key),
+			  m_aes_key_buf, strlen(p_test_vector->p_key));
+	iv_len = hex2bin(p_test_vector->p_iv, strlen(p_test_vector->p_iv),
+			 m_aes_iv_buf, strlen(p_test_vector->p_iv));
+	ad_len = hex2bin(p_test_vector->p_ad, strlen(p_test_vector->p_ad),
+			 m_aes_temp_buf, strlen(p_test_vector->p_ad));
+
+	if (encrypt) {
+		input_len = hex2bin(p_test_vector->p_plaintext,
+				    strlen(p_test_vector->p_plaintext),
+				    m_aes_input_buf,
+				    strlen(p_test_vector->p_plaintext));
+		output_len = hex2bin(p_test_vector->p_ciphertext,
+				     strlen(p_test_vector->p_ciphertext),
+				     m_aes_expected_output_buf,
+				     strlen(p_test_vector->p_ciphertext));
+	} else {
+		input_len = hex2bin(p_test_vector->p_ciphertext,
+				    strlen(p_test_vector->p_ciphertext),
+				    m_aes_input_buf,
+				    strlen(p_test_vector->p_ciphertext));
+		output_len = hex2bin(p_test_vector->p_plaintext,
+				     strlen(p_test_vector->p_plaintext),
+				     m_aes_expected_output_buf,
+				     strlen(p_test_vector->p_plaintext));
+	}
+}
+
+static void aes_setup_cbc_mac(void)
+{
+	aes_cbc_mac_clear_buffers();
+
+	static int i;
+	p_test_vector =
+		ITEM_GET(test_vector_aes_cbc_mac_data, test_vector_aes_t, i++);
+
+	unhexify_aes();
+}
+
+/**@brief Function for the AES MAC test execution.
+ */
+void exec_test_case_aes_cbc_mac(void)
+{
+	int err_code = -1;
+
+	mbedtls_cipher_context_t ctx;
+
+	err_code = cipher_init(&ctx, key_len, p_test_vector->mode);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	err_code = cipher_set_key(&ctx, key_len, MBEDTLS_ENCRYPT);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	err_code = cipher_set_padding(&ctx, p_test_vector->padding);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	start_time_measurement();
+	LOG_DBG("Using CBC for MAC (CBC MAC).");
```

Comment:
Log should be before start_time_measurement()

Reply:
Moved in 
a69739b